### ISSUE TITLE
14.2.1 - component up to date

### ISSUE THREAD

**Issue by elarlang** [OWASP/ASVS#2164]
Spin-off from #2088

Current requirement:

| # | Description | L1 | L2 | L3 | CWE |
| :---: | :--- | :---: | :---: | :---: | :---: |
| **14.2.1** | Verify that all components are up to date, preferably using a dependency checker during build or compile time. | ✓ | ✓ | ✓ | 1026 |

Some comments to keep ind mind:

* https://github.com/OWASP/ASVS/issues/2088#issuecomment-2355412099
* https://github.com/OWASP/ASVS/issues/2088#issuecomment-2362818721
* https://github.com/OWASP/ASVS/issues/2088#issuecomment-2362912214

My proposal:

| # | Description | L1 | L2 | L3 | CWE |
| :---: | :--- | :---: | :---: | :---: | :---: |
| **14.2.1** | Verify that all components are up to date. Note that updating only to patch publically known vulnerabilities relies on confidence that the vendor does not fix security holes without public announcements. | ✓ | ✓ | ✓ | 1026 |

The "Note" part can be also part of the chapter text.

**Comment by jmanico** [comment]
There are significant unpublished security vulnerabilities. I hate to suggest it because it’s hard work, but I’d rather see folks er on the side of patching even if the SCA scanners say it’s clear. Especially server side. Less so client side.

**Comment by elarlang** [comment]
What is the point in your comment, what it suggests what is not suggested?

**Comment by jmanico** [comment]
I do not think we should suggest _not patching unless its a known published vuln_ for L3. I think this requirement should be removed.

**Comment by elarlang** [comment]
What exact part of my proposal says that?

**Comment by jmanico** [comment]
This sentence implies it. And "publically" is spelled wrong (should be "publicly")

_Note that updating only to patch publically known vulnerabilities relies on confidence that the vendor does not fix security holes without public announcements._

If that is not what you meant I suggest clarifying it.

**Comment by elarlang** [comment]
The requirement says that you need to be up to date. The note points out, that if you update only for patched for **publicly** known issues, it contains security risk.

**Comment by jmanico** [comment]
Let me suggest clarifying it so its more obvious.

_Note that while being up-to-date with patches is essential, relying solely on updates for publicly disclosed vulnerabilities introduces risk, as vendors may fix security issues without public announcements._

**Comment by jmanico** [comment]
I think that is very reasonable!

**Comment by elarlang** [comment]
Re-open fpr moving it to V10.6
